common/src/logger.rs

```diff
@@ -1,71 +1,45 @@
-use bootloader_api::info::{FrameBufferInfo, PixelFormat};
+use crate::{framebuffer::FrameBufferWriter, serial::SerialPort};
+use bootloader_api::{config::LoggerStatus, info::FrameBufferInfo};
 use conquer_once::spin::OnceCell;
-use core::{
-    fmt::{self, Write},
-    ptr,
-};
-use font_constants::BACKUP_CHAR;
-use noto_sans_mono_bitmap::{
-    get_raster, get_raster_width, FontWeight, RasterHeight, RasterizedChar,
-};
+use core::fmt::Write;
 use spinning_top::Spinlock;
 
 /// The global logger instance used for the `log` crate.
 pub static LOGGER: OnceCell<LockedLogger> = OnceCell::uninit();
 
-/// A [`Logger`] instance protected by a spinlock.
-pub struct LockedLogger(Spinlock<Logger>);
-
-/// Additional vertical space between lines
-const LINE_SPACING: usize = 2;
-/// Additional horizontal space between characters.
-const LETTER_SPACING: usize = 0;
-
-/// Padding from the border. Prevent that font is too close to border.
-const BORDER_PADDING: usize = 1;
-
-/// Constants for the usage of the [`noto_sans_mono_bitmap`] crate.
-mod font_constants {
-    use super::*;
-
-    /// Height of each char raster. The font size is ~0.84% of this. Thus, this is the line height that
-    /// enables multiple characters to be side-by-side and appear optically in one line in a natural way.
-    pub const CHAR_RASTER_HEIGHT: RasterHeight = RasterHeight::Size16;
-
-    /// The width of each single symbol of the mono space font.
-    pub const CHAR_RASTER_WIDTH: usize = get_raster_width(FontWeight::Regular, CHAR_RASTER_HEIGHT);
-
-    /// Backup character if a desired symbol is not available by the font.
-    /// The '�' character requires the feature "unicode-specials".
-    pub const BACKUP_CHAR: char = '�';
-
-    pub const FONT_WEIGHT: FontWeight = FontWeight::Regular;
-}
-
-/// Returns the raster of the given char or the raster of [`font_constants::BACKUP_CHAR`].
-fn get_char_raster(c: char) -> RasterizedChar {
-    fn get(c: char) -> Option<RasterizedChar> {
-        get_raster(
-            c,
-            font_constants::FONT_WEIGHT,
-            font_constants::CHAR_RASTER_HEIGHT,
-        )
-    }
-    get(c).unwrap_or_else(|| get(BACKUP_CHAR).expect("Should get raster of backup char."))
-}
+/// A [`FrameBufferWriter`] instance protected by a spinlock.
+pub struct LockedLogger(
+    Spinlock<FrameBufferWriter>,
+    Spinlock<SerialPort>,
+    LoggerStatus,
+    LoggerStatus,
+);
```

Comment:
I would prefer to make this a normal struct with named fields. Also, we could directy resolve the `LoggerStatus` and set the fields as an `Option`:

```suggestion
/// A logger instance protected by a spinlock.
pub struct LockedLogger {
    framebuffer: OptionSpinlock<FrameBufferWriter>>,
    serial: Option<Spinlock<SerialPort>>,
}
```